doc/guides/V8-api-for-node.md

```diff
@@ -0,0 +1,210 @@
+## V8 API usage for Node.js
+
+v8 docs, not particularly useful:
+- https://v8.dev/docs
+
+Some API info at https://v8.dev/docs/embed
+
+- An isolate is a VM instance with its own heap.
+  Node has one isolate.
+  Can get from `Local<Object>->GetIsolate()`, `context->GetIsolate()`
+
+- `Local<...>`: A local handle is a pointer to an object. All V8 objects are
+  accessed using handles, they are necessary because of the way the V8 garbage
+  collector works.  Local handles can only be allocated on the stack, in a
+  scope, not with new, and will be deleted as stack unwinds, usually with a
+  scope.
+  - XXX don't see us create scopes much in node, why?
+  - XXX locals go into a HandleScope, for auto-dispose? Always, or by default?
+- Persistent handles last past C++ functions.
+  - PersistentBase::SetWeak trigger a callback from the garbage collector when
+    the only references to an object are from weak persistent handles.
+  - A UniquePersistent<SomeType> handle relies on C++ constructors and
+    destructors to manage the lifetime of the underlying object.
+  - A Persistent<SomeType> can be constructed with its constructor, but must be
+    explicitly cleared with Persistent::Reset.
```

Comment:
This is a good example of how usage is shifting – we introduced `node::Persistent<>`, which automatically resets in the destructor, but which becomes unnecessary through `v8::Global<>`, which also does that and additionally supports move semantics. (I’d also use `v8::Global<>` instead of `UniquePersistent` above, the latter is an alias for the former and I don’t think we use `UniquePersistent` in our codebase.)